### DISCUSSION TITLE
NavigationContainer nesting

### DISCUSSION THREAD

**Discussion by dsrenesanse** [react-navigation/react-navigation#11924] [Q&A]
Hi, I am trying to implement fully declarative navigation in my project and it's not really clear to me what is the purpose for NavigationContainer, because I can use NavigationContainer only once at the root of app, but nesting with independent flag working the same. The use case: nested stack navigators for different user flows.
Please tell what is the main difference between having only one NavigationContainer at the root or several nested.

**Comment by satya164** [comment]
https://reactnavigation.org/docs/navigation-container/#independent

> Setting it to `true` disconnects any children navigators from the parent container and doesn't allow navigation between them.
